user-interviews/ui-design-system
# uses yalc for linking the DS to RS

This will allow local development of the DS against RS in fewer steps and require less shuffling.  It will also prepare us better for using Typescript in the DS, as `yarn link` runs into some strange behavior in RS with regard to the types we have for react:  
<img width="745" alt="Screenshot 2023-09-06 at 4 05 08 PM" src="https://github.com/user-interviews/ui-design-system/assets/3161151/59789d55-a5f5-4008-a6cb-6b7c532b197b">


### Manual Test plan
[//]: # (
How did you test this manually? How can your reviewer test the workflow?
If this is a bug, how can you reproduce the bug?
)

1. Checkout this branch
2. Checkout the corresponding RS branch: `infra/yalc-setup`
3. run `yarn rs:link:watch` from the root of the DS
4. run `bin/shakapacker-dev-server` from the root of the RS
5. make changes to a DS component

Expected outcome:
- changes are reflected in RS
- webpack recompiles
- browser refreshes to reflect changes in DS

Run `yarn rs:unlink` once testing is complete.


Corresponding Rails PR: https://github.com/user-interviews/rails-server/pull/20684
